crates/blockchain/blockchain.rs

```diff
@@ -206,5 +307,94 @@ fn get_total_blob_gas(tx: &EIP4844Transaction) -> u64 {
     GAS_PER_BLOB * tx.blob_versioned_hashes.len() as u64
 }
 
+fn total_difficulty_check<'a>(
+    head_block_hash: &'a H256,
+    head_block: &'a BlockHeader,
+    storage: &'a Store,
+) -> Result<(), InvalidForkChoice> {
+    if !head_block.difficulty.is_zero() || head_block.number == 0 {
+        let total_difficulty = storage
+            .get_block_total_difficulty(*head_block_hash)
+            .map_err(wrap)?;
+        let parent_total_difficulty = storage
+            .get_block_total_difficulty(head_block.parent_hash)
+            .map_err(wrap)?;
+        let terminal_total_difficulty = storage
+            .get_chain_config()
+            .map_err(wrap)?
+            .terminal_total_difficulty;
+        if terminal_total_difficulty.is_none()
+            || total_difficulty.is_none()
+            || head_block.number > 0 && parent_total_difficulty.is_none()
```

Comment:
Seems like we have multiple None checks and then we use the values if they're Some, right?

How about using ok_or to do an early return at the top-most level of this function?

Take for example total_difficulty:

```rust
  let total_difficulty = storage


             .get_block_total_difficulty(*head_block_hash)
             .map_err(wrap)?
             .ok_or(//Error) // Return error if None, if Some, keep the inner value.
```